### ISSUE TITLE
NUnit1001: False positive

### ISSUE THREAD

**Issue by jcurl** [nunit/nunit.analyzers#162]
Downloaded 0.1.0-dev-00205, analyzed my solution and I get a false positive for NUnit1001 when the types have an implicit conversion available.

I'm using NUnit 2.6.7, but I don't expect this to be a problem in NUnit3 either.

The error reports (for both these cases):
"The value of the argument at position '0' cannot be assigned to the argument 'value'"

```csharp
[TestCase(uint.MaxValue, "4294967295")]
[TestCase(0x8FFFFFFF, "2415919103")]
public void ClsCompliantCtor(long value, string output) {
 ...
}
```

**Comment by mikkelbu** [comment]
Thanks for the bug report @jcurl. I can confirm the problem, but I did not think that the analyzers also worked with NUnit 2.X.Y 😄.

We can probably get inspiration from how the NUnit code handles implicit conversions https://github.com/nunit/nunit/blob/46f8b2ae280a6a6150e0f0bf86718a783d8b0c9d/src/NUnitFramework/framework/Internal/Reflect.cs#L195-L206

**Comment by stevenaw** [comment]
Interesting, perhaps there's an inconsistency in the framework. I think ParamAttributeTypeConversions.cs maps those parameter types a bit differently than Reflect.cs

https://github.com/nunit/nunit/blob/46f8b2ae280a6a6150e0f0bf86718a783d8b0c9d/src/NUnitFramework/framework/Internal/ParamAttributeTypeConversions.cs#L119

**Comment by mikkelbu** [comment]
I'm think I was too quick. The `convertibleValueTypes` map is used to find constructors that match, and I think in that setting we don't have quite the same translation as for tests. Googling it seems we can use `Compilation.HasImplicitConversion` - see e.g. dotnet/roslyn#26719 to determine if one type can be implicitly converted into another.

**Comment by jcurl** [comment]
I'm not familiar with Roslyn Compiler as a Service and analyzers (sounds interesting!) but will https://github.com/dotnet/roslyn/pull/26719 imply a minimum version? I'm using VS2017.

**Comment by Dreamescaper** [comment]
I'll take a look into this.
